lsp-mode.el

```diff
@@ -4587,11 +4591,13 @@ remote machine and vice versa."
     (-when-let (matching-clients (->> lsp-clients
                                       hash-table-values
                                       (-filter (-lambda (client)
-                                                 (and (or (-some-> client lsp--client-activation-fn (funcall buffer-file-name buffer-major-mode))
-                                                          (and (not (lsp--client-activation-fn client))
-                                                               (-contains? (lsp--client-major-modes client) buffer-major-mode)
-                                                               (eq (---truthy? remote?) (---truthy? (lsp--client-remote? client)))))
-                                                      (-some-> client lsp--client-new-connection (plist-get :test?) funcall))))))
+                                                 (and (or (eq lsp-enabled-clients nil)
```

Comment:
can you move this as a second condition and add logging that the client is not enabled? E. g.
1. Check whether it is present, matches mode, etc.
2. Check whether it is enabled
3. If it is matching and not enabled print a message similar to what we do for blacklisted dirs
https://github.com/emacs-lsp/lsp-mode/blob/42b1e5c1f82e56734777529fadba9291e8a9dc15/lsp-mode.el#L4906